images/virtualization-artifact/pkg/controller/vmclass/vmclass_reconciler.go

```
@@ -93,15 +95,31 @@ func (r *Reconciler) SetupController(ctx context.Context, mgr manager.Manager, c
 					})
 					continue
 				}
-				if !annotations.MatchLabels(node.GetLabels(), class.Spec.NodeSelector.MatchLabels) {
-					continue
+
+				if len(class.Spec.NodeSelector.MatchLabels) != 0 {
+					if !annotations.MatchLabels(node.GetLabels(), class.Spec.NodeSelector.MatchLabels) {
+						continue
+					}
 				}
-				ns, err := nodeaffinity.NewNodeSelector(&corev1.NodeSelector{
-					NodeSelectorTerms: []corev1.NodeSelectorTerm{{MatchExpressions: class.Spec.NodeSelector.MatchExpressions}},
-				})
-				if err != nil || !ns.Match(node) {
-					continue
+
+				if len(class.Spec.NodeSelector.MatchExpressions) != 0 {
+					ns, err := nodeaffinity.NewNodeSelector(&corev1.NodeSelector{
+						NodeSelectorTerms: []corev1.NodeSelectorTerm{{MatchExpressions: class.Spec.NodeSelector.MatchExpressions}},
+					})
+					if err != nil || !ns.Match(node) {
```

Never ignore errors; otherwise, you'll have a hard time figuring out why the event didn't arrive. Please check all the errors in this file to ensure none are being ignored.